### PR TITLE
Remove flags, bunch of other stuff

### PR DESCRIPTION
**Pull request by fubarhouse**
* add gitignore
* add dockerignore
* remove input flags
* updated and tested docs

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -19,51 +20,38 @@ const (
...
+	PROXY_APP_USERNAME string
+	PROXY_APP_PASSWORD string
+	PROXY_APP_PORT string
 )
 
-func main() {
-
-	kingpin.Parse()
-
+func init() {
```
Comment (nickschuch) [inline main.go:29]: I would perfer to see a `main()` and `run()` approach if you want to initialize the Skpr client outside of `main()`.

File: main.go [hunk 2]
```diff
@@ -19,51 +20,38 @@ const (
 )
 
 var (
-	username = kingpin.Flag("username", "Username").String()
-	password = kingpin.Flag("password", "Password").String()
-	endpoint = kingpin.Flag("url", "URL to proxy service to, which includes the port.").Short('u').String()
-	port     = kingpin.Flag("port", "Port to expose the service to the host").Default("8080").Short('p').Int()
+	PROXY_APP_ENDPOINT string
```
Comment (nickschuch) [inline main.go:23]: Should be camel case. These could also move into `main()`.

File: main.go [hunk 3]
```diff
@@ -73,26 +61,27 @@ func main() {
...
-	fmt.Printf("Proxy configured to use port %s\n", portToUse)
+	fmt.Printf("Proxy configured to use port %s\n", PROXY_APP_PORT)
 	for _, target := range targets {
-		fmt.Printf("Starting proxy on http://localhost:%s for endpoint %s\n", portToUse, target.URL)
+		fmt.Printf("Starting proxy on http://localhost:%s for endpoint %s\n", PROXY_APP_PORT, target.URL)
+	}
+
+	// Handle Authentication
+	if PROXY_APP_USERNAME != "" && PROXY_APP_PASSWORD != "" {
+		e.Use(middleware.BasicAuth(func(username, password string, c echo.Context) (bool, error) {
```
Comment (nickschuch) [inline main.go:72]: Looks like this is setting up basic auth for the "client -> api" and not for the reverse proxy connection to the backend.

File: main.go [hunk 4]
```diff
@@ -73,26 +61,27 @@ func main() {
...
+			}
+			return false, nil
+		}))
 	}
 
 	// Start serving the proxy as configured.
 	e.Use(middleware.Proxy(middleware.NewRoundRobinBalancer(targets)))
-	localAddress := fmt.Sprintf(":%s", portToUse)
+	localAddress := fmt.Sprintf(":%s", PROXY_APP_PORT)
 	e.Logger.Fatal(e.Start(localAddress))
```
Comment (nickschuch) [inline main.go:85]: I wonder if `e.Start()` validates the `localAddress`. That way you could just use `os.Get("PROXY_APP_ADDRESS")` here and have a default in the Docker image.

File: main.go (hunk 1 above)
Comment (fubarhouse) [reply main.go]: Done - `Run()` is exported for external use.

File: main.go (hunk 2 above)
Comment (fubarhouse) [reply main.go]: Done.

File: main.go (hunk 3 above)
Comment (fubarhouse) [reply main.go]: Removed.

File: main.go (hunk 4 above)
Comment (fubarhouse) [reply main.go]: Reworked how the port is used - removed localAddress.
I have checked and no, no validation occurs.

**Comment by fubarhouse** [comment]
Just need to fix up the docker build via goreleaser now...

**Comment by fubarhouse** [comment]
This is now ready @nickschuch.

**Pull request merged**